backend-rust/src/indexer/statistics.rs

```diff
@@ -1,52 +1,66 @@
+use crate::scalar_types::DateTime;
 use anyhow::Result;
+use concordium_rust_sdk::base::contracts_common::CanonicalAccountAddress;
 use sqlx::{Postgres, Transaction};
+use std::collections::HashMap;
+use tracing::debug;
 
 #[derive(Debug, Clone, PartialEq, Eq, Hash)]
 pub(crate) enum BakerField {
     Added,
     Removed,
 }
 
-pub(crate) struct Statistics {
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub(crate) enum RewardField {
+    Account(CanonicalAccountAddress),
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub(crate) enum StatisticsField {
+    Baker(BakerField),
+    Reward(RewardField),
+}
```

Comment:
Instead of building a larger and larger enum to represent every type of increment and have a single increment function, it seems simpler to just introduce separate `increment` functions for each statistic. You avoid constructing these enums everywhere and branching in the increment function.

Reply:
I did this to avoid having annoying namings on the methods. 

Sure this is another way of doing

Reply:
But this is making something abstract for no reason, we don't need to be able to call a single `increment` function where the field is abstract, because we know which to increment in every place.
We only introduce complexity if strictly needed, and so far we don't need anything to be generic.